logprep/processor/generic_resolver/processor.py

```diff
@@ -78,47 +81,54 @@ def _apply_rules(self, event, rule):
-                                        event[key] = event.get("key", [])
-                                        if dest_val not in event[key]:
-                                            event[key].append(dest_val)
-                                    else:
-                                        event[key] = dest_val
-                                    break
-                                event[key] = {}
-                            if isinstance(event[key], dict):
-                                event = event[key]
-                            else:
-                                if rule.append_to_list and isinstance(event[key], list):
-                                    if dest_val not in event[key]:
-                                        event[key].append(dest_val)
-                                else:
-                                    conflicting_fields.append(keys[idx])
+                        success = self._add_uniquely_to_list(event, rule, target_field, dest_val)
+                        if not success:
+                            conflicting_fields.append(target_field)
 
+            # LIST
             for pattern, dest_val in rule.resolve_list.items():
-                if src_val and re.search(pattern, src_val):
-                    for idx, key in enumerate(keys):
-                        if key not in event:
-                            if idx == len(keys) - 1:
-                                if rule.append_to_list:
-                                    event[key] = event.get("key", [])
-                                    event[key].append(dest_val)
-                                else:
-                                    event[key] = dest_val
-                                break
-                            event[key] = {}
-                        if isinstance(event[key], dict):
-                            event = event[key]
-                        else:
-                            conflicting_fields.append(keys[idx])
+                if re.search(pattern, source_value):
+                    success = add_field_to(
+                        event,
+                        target_field,
+                        dest_val,
+                        extends_lists=rule.extend_target_list,
+                        overwrite_output_field=rule.overwrite_target,
+                    )
+                    if not success:
+                        conflicting_fields.append(target_field)
                     break
-
+        self._handle_missing_fields(event, rule, rule.field_mapping.keys(), source_values)
         if conflicting_fields:
             raise FieldExistsWarning(rule, event, conflicting_fields)
 
+    @staticmethod
+    def _add_uniquely_to_list(
+        event: dict,
+        rule: GenericResolverRule,
+        target: str,
+        content: Union[str, float, int, list, dict],
+    ) -> bool:
+        """Extend list if content is not already in the list"""
```

Comment:
I don't think it can be solved with a set, since list values could already exist in the event.
This could lead to duplicates.
A set would only ensure that newly added values are unique in regard to each other.